backend/src/main/java/hu/flowacademy/worksheet/service/UserService.java

```diff
@@ -117,4 +123,15 @@ public User setUserActivity(Long id, Status status) throws ValidationException {
         keycloakClientService.setUserStatus(toChange);
         return userRepository.save(toChange);
     }
+
+    public Optional<User> getCurrentUser() {
+        return Optional.ofNullable(SecurityContextHolder.getContext())
```

Comment:
Ez megdöbbentő próbálkozás, én szeretem a streameket, de ez egy stack halom, ahelyett, hogy leírnád ezt: String email = ((KeycloakPrincipal<KeycloakSecurityContext>) SecurityContextHolder.getContext().getAuthentication().getPrincipal()).getKeycloakSecurityContext().getToken().getEmail();

Reply:
Az én javaslatom volt, mert optional mapinggel kizárható a NullPoiinterException, bárhol is használná a függvényt, legfeljebb egy empty lesz a válasz.